index.js

```diff
@@ -983,131 +1047,205 @@
     return x => this (f (x));
   }
 
+  const staticImplementations = {
+    String: {
+      empty: String$empty,
+    },
+    Array: {
+      empty: Array$empty,
+      of: Array$of,
+      chainRec: Array$chainRec,
+      zero: Array$zero,
+    },
+    Object: {
+      empty: Object$empty,
+      zero: Object$zero,
+    },
+    Function: {
+      id: Function$id,
+      of: Function$of,
+      chainRec: Function$chainRec,
+    },
+  };
+
   const staticMethod = (name, typeRep) => {
-    switch (typeRep.name + '.' + name) {
-      case 'String.fantasy-land/empty':
-        return String$empty;
-      case 'Array.fantasy-land/empty':
-        return Array$empty;
-      case 'Array.fantasy-land/of':
-        return Array$of;
-      case 'Array.fantasy-land/chainRec':
-        return Array$chainRec;
-      case 'Array.fantasy-land/zero':
-        return Array$zero;
-      case 'Object.fantasy-land/empty':
-        return Object$empty;
-      case 'Object.fantasy-land/zero':
-        return Object$zero;
-      case 'Function.fantasy-land/id':
-        return Function$id;
-      case 'Function.fantasy-land/of':
-        return Function$of;
-      case 'Function.fantasy-land/chainRec':
-        return Function$chainRec;
-      default:
-        return null;
+    switch (typeRep) {
+      case String: return staticImplementations.String[name];
+      case Array: return staticImplementations.Array[name];
+      case Object: return staticImplementations.Object[name];
+      case Function: return staticImplementations.Function[name];
+    }
+
+    const prefixedName = 'fantasy-land/' + name;
+    if (typeof typeRep[prefixedName] === 'function') {
+      return typeRep[prefixedName];
+    }
```

Comment:
This means this function now throws on inaccessible inputs (like `null`), where before it would return `null`. Since the return value of this function is used by `test` to determine if something has a typeclass, that functionality might've broken.

Reply:
Are you saying that `typeRep` can be `null`?

Reply:
I can't find a way to make it so, but there used to be code in place to deal with that.

Reply:
Correction. I can't find a way to make it so in places where no exception is expected. I can easily make it so when an exception would've occurred anyway:

```js
> Z.of(null, 42)
Uncaught TypeError: Cannot read property 'fantasy-land/of' of null
    at staticMethod
```

Reply:
Ah, the problem I foresaw here is already guarded against:

https://github.com/sanctuary-js/sanctuary-type-classes/pull/151/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R221

See `x != null`.